account_payment_order_inter_company/models/account_payment_order.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2022 Akretion France (http://www.akretion.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class AccountPaymentOrder(models.Model):
+    _inherit = "account.payment.order"
+
+    def _prepare_move_line_vals(self, payment_line, invoice, journal, company):
+        vals_list = []
+        vals1 = {}
+        vals2 = {}
+        vals = {
+            "journal_id": journal.id,
+            "move_id": invoice.id,
```

Comment:
Warning you are adding new line in the invoice move !!
you need to create a new move with all payment line